language/stdlib/modules/DesignatedDealer.move

```diff
@@ -89,7 +89,7 @@ module DesignatedDealer {
         // If the tier_index is one past the bounded tiers, minting is unbounded
         let number_of_tiers = Vector::length(tiers);
         let tier_check = &mut false;
-        if (tier_index == number_of_tiers) {
+        if (tier_index >= number_of_tiers) {
```

Comment:
Why did you change it to `>=`?

Reply:
Currently, this is not an issue due to (1) the assert i added (2) the fact that 4 tiers are created on setup. But previously if the tier_index was greater than the number of tiers then we would enter the else loop and access an invalid index of the tier list (which was causing the NATIVE_FN_ERROR). This >= should prevent this from happening in case (1) or (2) are changed in the future.

Reply:
I definitely agree that we don't want to enter the `else` clause here.
Just thought that there should be a better way to handle it, may be with separate branches explicitly checking `>` and `==` conditions (the current solution is introducing a potential loophole in case the function is not called by `tiered_mint`)